Automattic/wp-calypso
# Launchpad: Launch Celebration Modal

### Time Estimate
Review: Mid
Test: Short

### Proposed Changes
This PR adds a new modal celebrating launching a site

#### Dev Notes
I am currently using the `CelebrateSiteLaunch` component to render the modal which I suspect it's not ideal. I would love some suggestions on whether we should make the modal its own component and ideas on how to hook into the site launch and know when to show the modal.
<img width="1113" alt="image" src="https://user-images.githubusercontent.com/20927667/216691170-b1417605-e8f9-429d-9ebb-6a0f67fca76e.png">

### Testing
1. Checkout this branch
2. Go to http://calypso.localhost:3000/setup/free/intro or any other Launchpad flow
3. Create & Launch a site
4. Once you are redirected to My Home you will see the modal pop-up. Verify the `?celebrateLaunch=true` is appended to the URL.
5. Make sure it looks good according to the mockup (you can find it in the thread linked below)
6. As soon as the modal opens verify the `?celebrateLaunch=true` is removed from the URL without the page being reloaded
7. Visit Site button should redirect you to the site
8. Customize domain should redirect you to the domain screen. When you go back you should go to my home without seeing the modal again (no `?celebrateLaunch=true` in the URL)
9. Clicking the close button, pressing Escape or clicking outside the modal should close the modal


https://user-images.githubusercontent.com/20927667/216691284-c041de5f-af18-4ccc-ac59-0d7b55e87c5b.mov



### Project Thread
Related to #72021 
p9Jlb4-6dJ-p2
